### PR TITLE
Add SDL2 support

### PR DESCRIPTION
**Pull request by deveee**
If you build it, make sure that _IRR_COMPILE_WITH_SDL2_DEVICE_ is enabled either as a compiler flag or in IrrCompileConfig.h

As far as I tested, Irrlicht examples work fine with SDL2 on Linux, Android, Windows and MacOS. Supported drivers are OpenGL, GLES2 and GLES1.

Compared to other drivers, one missing feature is custom mouse cursor support.

Currently Visual Studio, Xcode, Codeblocks, MinGW projects are not modified, so you have to manually add SDL library and include directories to the project.

### CODE REVIEW

**Comment by MoNTE48** [comment]
I think [Add example for SDL2 on Android](https://github.com/MoNTE48/Irrlicht/pull/1/commits/6209f0a605bec725664229e233f98c4dc41872a6) commit can be dropped.

**Comment by MoNTE48** [comment]
@deveee After this commit https://github.com/MoNTE48/Irrlicht/pull/1/commits/c9dfd658c89ce00d41b2da03bc78db019325c7a5 I get the next error when starting MultiCraft:
error build: Undefined symbol: irr::CIrrDeviceMacOSX::CIrrDeviceMacOSX(irr::SIrrlichtCreationParameters const&)

**Comment by deveee** [comment]
Because both SDL2 and OSX devices were built (it's possible to select irrlicht device using DriverType in creation params). This should disable OSX device if compiled with SDL2 (I can test later).
```
diff --git a/include/IrrCompileConfig.h b/include/IrrCompileConfig.h
index 487f35c1..fd484bd0 100644
--- a/include/IrrCompileConfig.h
+++ b/include/IrrCompileConfig.h
@@ -119,7 +119,9 @@
 #define NO_IRR_COMPILE_WITH_OPENGL_
 #else
 #define _IRR_OSX_PLATFORM_
+#if !defined(_IRR_COMPILE_WITH_SDL2_DEVICE_)
 #define _IRR_COMPILE_WITH_OSX_DEVICE_
+#endif
 #define NO_IRR_COMPILE_WITH_OGLES1_
 #define NO_IRR_COMPILE_WITH_OGLES2_
 #define NO_IRR_COMPILE_WITH_WEBGL1_
```

File: source/Irrlicht/Irrlicht.cpp [hunk 1]
```diff
@@ -73,6 +73,11 @@ namespace irr
 
 		IrrlichtDevice* dev = 0;
 
+#ifdef _IRR_COMPILE_WITH_SDL_DEVICE_
+		if (params.DeviceType == EIDT_SDL || (!dev && params.DeviceType == EIDT_BEST))
+			dev = new CIrrDeviceSDL(params);
+#endif
+
```
Comment (MoNTE48) [inline source/Irrlicht/Irrlicht.cpp]: Can you not move it up?

**Pull request merged**